### PR TITLE
Edge Security Manual

### PR DESCRIPTION
**Pull request by pradumnnec**

### CODE REVIEW

File: doc/en/source/https.rst [hunk 1]
```diff
@@ -209,7 +209,7 @@ Terminology
```
Comment (naveennec) [inline doc/en/source/https.rst:209]: [Minor]--Security Architecture
1- both users are different better to give different name to them.
2- Rewrite the naming convention to segregate with each other-- write below suggestion name as discussed Generate user 1 token, Generate user 2 token, Device 1 token, Device 2 token, Device 3 token, Device 4 token
3- Write sequencing to show to flow

**Review by naveennec — CHANGES_REQUESTED** [review]

File: doc/en/source/https.rst [hunk 2]
```diff
@@ -227,11 +227,11 @@ FogFlow cloud node flow:
 FogFlow edge node flow:
 
 
-1. On behalf of edge node, one application will be register on keyrock it will be using OAuth credentials. Detail explanation is given in below topics of this tutorial. Click `here`_ to refer.
+1. On behalf of edge node, one instance of PEP Proxy will be pre-registered on keyrock, edge will be using oauth credentials to fetch PEP Proxy  details. Detail explanation is given in below topics of this tutorial. Click `here`_ to refer.
 
 2. After the authentication edge node will be able to communicate with FogFlow cloud node.
 
-3. Any device can register itself or communicate with FogFlow edge node using edge’s OAuth access-token.
+3. Any device can register itself or communicate with FogFlow edge node using  access-token generated on behalf of each IoT sensor registered at Keyrock.
```
[the remark this answers is not in the document]
Comment (pradumnnec) [reply doc/en/source/https.rst]: Incorporated Comment

File: doc/en/source/https.rst (hunk 1 above)
Comment (pradumnnec) [reply doc/en/source/https.rst:209]: 1. Security Architecture png is updated.
2. Sequencing also done.

File: doc/en/source/https.rst [hunk 3]
```diff
@@ -209,29 +209,67 @@ Terminology
...
+   	"http://KEYROCK_IP_ADDRESS:3000/v1/auth/tokens" 
+
+ 
+
+**Step 2** : IDM will return an access token to user, based on his credentials in rquest made in previous step.
 
+**Step 3** : User shares his access token ( i.e. User Access Token ) to IoT Device.
 
-FogFlow cloud node flow:
+**Step 4** : IoT Device is regstered with IDM using User Access Token as shown in `below`_ topics. As an output, IDM provides an Id and password to IoT Device based on the registration request.
```
Comment (naveennec) [inline doc/en/source/https.rst:242]: Then IOT devices get registered using the user access token and  in response it get device credentials(ID and password) as shown in `below`_ topics

File: doc/en/source/https.rst [hunk 4]
```diff
@@ -209,29 +209,67 @@ Terminology
...
+
+.. _`below`: https://fogflow.readthedocs.io/en/latest/https.html#register-iot-device-on-keyrock
+
+**Step 5** : Using the above ID and password for IoT Device (generated in previous request), a request is made to IDM to generate an access token specific for that Device only as shown in `later`_ section.
```
Comment (naveennec) [inline doc/en/source/https.rst:246]: IOT Device then make access token request using above Device credentials and in response it get Device access token,

File: doc/en/source/https.rst [hunk 5]
```diff
@@ -209,29 +209,67 @@ Terminology
...
+
+.. _`later`: https://fogflow.readthedocs.io/en/latest/https.html#register-iot-device-on-keyrock
+
+**Step 6** : Now, using the above access token, the IoT Device can interact with Edge node via making Fogflow specific requests to PEP Proxy port.
```
Comment (naveennec) [inline doc/en/source/https.rst:250]: better to write "using Device access token."

**Review by naveennec — CHANGES_REQUESTED** [review]

File: doc/en/source/https.rst [hunk 6]
```diff
@@ -209,29 +209,67 @@ Terminology
...
 
 
+Note: Red markers represent IoT Device interaction with cloud and  green markers represent IoT Device interaction with Edge Node
 
-Note: In above diagram n ( n is numeric) is used for cloud node and n’ for edge node.
+ 
+
+**Flow of Requests as shown in diagram:**
+
+**Step 1** : User will make a request to IDM using his credentials to generate access token specific for that user. For example:
```
[the remark this answers is not in the document]
Comment (pradumnnec) [reply doc/en/source/https.rst]: Comment Incorporated.

File: doc/en/source/https.rst [hunk 7]
```diff
@@ -209,29 +209,67 @@ Terminology
...
+        --header "Content-Type: application/json" \
+        --data-binary "{
+     	\"name\": \"admin@test.com\",
+     	\"password\": \"1234\"
+   	}" \
+   	"http://KEYROCK_IP_ADDRESS:3000/v1/auth/tokens" 
+
+ 
+
+**Step 2** : IDM will return an access token to user, based on his credentials in rquest made in previous step.
```
[the remark this answers is not in the document]
Comment (pradumnnec) [reply doc/en/source/https.rst]: Comment Incorporated.

File: doc/en/source/https.rst (hunk 3 above)
Comment (pradumnnec) [reply doc/en/source/https.rst]: Comment Incorporated.

File: doc/en/source/https.rst (hunk 4 above)
Comment (pradumnnec) [reply doc/en/source/https.rst]: Comment Incorporated.

File: doc/en/source/https.rst (hunk 5 above)
Comment (pradumnnec) [reply doc/en/source/https.rst]: Comment Incorporated.

**Pull request merged**